server/routes/dates.js

```diff
@@ -0,0 +1,111 @@
+const express = require('express')
+const moment = require('moment')
+
+const cardData = require('../db/cardData')
+const graph = require('../db/graph')
+const activities = require('../db/activities')
+
+const router = express.Router()
+
+module.exports = router
+
+// router.get('/cards', (req, res) => {
+//   cardData.getCards()
+// })
+
+router.post('/', (req, res) => {
+  // req.body required to look like :
+  // {userId: 1, records:[{activityId:1, rating:1, log:'blah'},{more cards}]}
+  const {userId, records} = req.body
+
+  // if (isRecordOkay) {
+  cardData.addDate({user_id: userId})
+    .then(dateId => {
+      for (let rec of records) {
+        rec.date_id = dateId[0]
+        rec.activity_id = rec.activityId
+        delete rec.activityId
+      }
+
+      cardData.addRecords(records)
+        .then(() => {
+          // res.json(records)
+          res.json({Okay: true})
+        })
+        .catch((err) => res.json({Okay: false, error: err.message}))
```

Comment:
It's a good idea to specify which HTTP status code you want to send back to the client, especially for errors. Without it, there's a risk that Superagent will think everything's fine, and won't reject the promise... so your error will not end up in the `.catch` block (in the async action creator). So something like:

```js
  .catch(err => res.status(500).json({ ok: false, error: err.message })
```